packages/nextjs/src/server/__tests__/content-security-policy.test.ts

```diff
@@ -0,0 +1,357 @@
+    });
+
+    it('should generate unique nonces', () => {
+      const nonce1 = generateNonce();
+      const nonce2 = generateNonce();
+      expect(nonce1).not.toBe(nonce2);
+    });
+  });
+
+  describe('createCSPHeader', () => {
+    const testHost = 'example.com';
+
+    it('should create a standard CSP header with default directives', () => {
+      const result = createCSPHeader('standard', testHost);
+
+      const directives = result.header.split('; ');
+
+      expect(directives).toContainEqual("default-src 'self'");
+      expect(directives).toContainEqual("connect-src 'self' https://api.stripe.com https://maps.googleapis.com *.clerk.accounts.dev clerk.example.com");
+      expect(directives).toContainEqual("form-action 'self'");
+      expect(directives).toContainEqual("frame-src 'self' https://challenges.cloudflare.com https://*.js.stripe.com https://js.stripe.com https://hooks.stripe.com");
+      expect(directives).toContainEqual("img-src 'self' https://img.clerk.com");
+      expect(directives).toContainEqual("style-src 'self' 'unsafe-inline'");
+      expect(directives).toContainEqual("worker-src 'self' blob:");
+
+      // script-src can vary based on env, so check each value separately
+      const scriptSrc = directives.find(d => d.startsWith('script-src'));
+      expect(scriptSrc).toBeDefined();
+      expect(scriptSrc).toContain("'self'");
+      expect(scriptSrc).toContain("'unsafe-inline'");
+      expect(scriptSrc).toContain('https:');
+      expect(scriptSrc).toContain('http:');
+      expect(scriptSrc).toContain('https://*.js.stripe.com');
+      expect(scriptSrc).toContain('https://js.stripe.com');
+      expect(scriptSrc).toContain('https://maps.googleapis.com');
+      if (process.env.NODE_ENV !== 'production') {
+        expect(scriptSrc).toContain("'unsafe-eval'");
+      }
+
+      expect(result.nonce).toBeUndefined();
+    });
+
+    it('should create a strict-dynamic CSP header with nonce', () => {
+      const result = createCSPHeader('strict-dynamic', testHost);
+
+      // Extract the script-src directive and verify it contains the required values
+      const directives = result.header.split('; ');
+      const scriptSrcDirective = directives.find((d: string) => d.startsWith('script-src')) ?? '';
+      expect(scriptSrcDirective).toBeDefined();
+
+      const scriptSrcValues = scriptSrcDirective.replace('script-src ', '').split(' ');
+      expect(scriptSrcValues).toContain("'self'");
+      expect(scriptSrcValues).toContain("'unsafe-inline'");
+      expect(scriptSrcValues).toContain("'strict-dynamic'");
+      expect(scriptSrcValues.some(val => val.startsWith("'nonce-"))).toBe(true);
+
+      expect(result.nonce).toBeDefined();
+      expect(result.nonce).toMatch(/^[A-Za-z0-9+/=]+$/);
+    });
+
+    it('should handle custom directives as an object', () => {
+      const customDirectives = {
+        'default-src': ["'none'"],
+        'img-src': ["'self'", 'https://example.com'],
```

Comment:
I ended up writing some crazy logic in my implementation that allowed values that needed to be quoted to be passed in normally, and the library would automatically quote them if necessary. Happy to pass that over here if that's useful, the rules around what should and shouldn't be quoted aren't exactly clear to everyone getting into CSPs and it looks weird to double quote values like this

Reply:
Removed those single quotes from the tests. It's handled automatically so it will accept both quoted/unquoted and output the correct format